platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -343,7 +346,7 @@ class DefaultConfigurationCache internal constructor(
         stoppable.addIfInitialized(lazyBuildTreeModelSideEffects)
         stoppable.addIfInitialized(lazyIntermediateModels)
         stoppable.addIfInitialized(lazyProjectMetadata)
-        stoppable.add(store)
+        stoppable.addIfInitialized(storeDelegate)
```

Comment:
Don't try to create the store if closing after an error to initialize it or some other state.